src/sql/workbench/browser/actions.ts

```diff
@@ -20,9 +20,13 @@ import { IAccountManagementService } from 'sql/platform/accounts/common/interfac
 import { ILogService } from 'vs/platform/log/common/log';
 import { IInsightsConfig } from 'sql/platform/extensions/common/extensions';
 
-export interface BaseActionContext {
-	object?: ObjectMetadata;
-	profile?: IConnectionProfile;
+export interface BaseActionContext extends azdata.ConnectedContext {
+	object?: azdata.ObjectMetadata;
+	/**
+	 * Override connectionProfile from ConnectedContext
+	 * with IConnectionProfile type
+	 */
+	connectionProfile?: IConnectionProfile | undefined;
```

Comment:
I didn't notice this was using the "internal" IConnectionProfile...we can't do this because otherwise any internal actions which expect the internal version with the extra functions/properties could fail since those wouldn't actually exist.

My suggestion would be to remove this override, then for any handlers which need the extra stuff have it create an internal ConnectionProfile from the azdata.IConnectionProfile passed in. It's not pretty but that should at least be safe...

Reply:
Alternatively - if that ends up being a bunch of work then I think it's fine to just leave the current behavior as is. The command stuff is a pretty hidden feature and probably not worth spending a bunch of time fixing it unless we start having more people using it.

Reply:
Well, that a very heavy dependency.. I'll let it be as-is then..

Reply:
😭 Yeah, it's one of the things I really wish we had time/resources to go in and refactor. The many different variations of connection profiles and how they're all used is extremely confusing and easy to mess up...